napari/layers/_shapes_layer/model.py

```diff
@@ -1148,7 +1233,142 @@ def to_xml_list(self, shape_type=None):
+        argument leads to only labels from that particular `shape_type` being
+        returned. These labels will be renumbered appropriately. For
+        overlapping shapes z-ordering will be respected.
+
+        Parameters
+        ----------
+        labels_shape : np.ndarray | tuple | None
+            Tuple defining shape of labels image to be generated. If non
+            specified, takes the max of all the vertiecs
+        shape_type : {'line', 'rectangle', 'ellipse', 'path', 'polygon'} |
+                     None, optional
+            String of shape type to be included.
+
+        Returns
+        ----------
+        labels : np.ndarray
+            Integer array where each value is either 0 for background or an
+            integer up to N for points inside the corresponding shape.
+        """
+        if labels_shape is None:
+            labels_shape = self.shape
+
+        if self.ndim == 2:
+            # For 2D shapes just convert current view to labels
+            labels = self._data_view.to_labels(labels_shape=labels_shape,
+                                               shape_type=shape_type)
+        elif self.broadcast:
+            # For broadcast shapes convert current view to labels and
+            # broadcast across sliced dimensions
+            labels = self._data_view.to_labels(labels_shape=labels_shape[-2:],
+                                               shape_type=shape_type)
+            labels = np.broadcast_to(labels, labels_shape)
+        else:
+            # For nD insert each keyed slice into correctd place in volume
+            # and increment integer label of shape
+            labels = np.zeros(labels_shape)
+            nshapes = 0
+            for key, data in self.data.items():
+                if len(key) > 0:
+                    slices = data.to_labels(labels_shape=labels_shape[-2:],
+                                            shape_type=shape_type)
+                    slices[slices > 0] = slices[slices > 0] + nshapes
+                    labels[key] = slices
+                    nshapes = nshapes + len(data.shapes)
+        return labels
+
+    def to_list(self, shape_type=None):
+        """Returns the vertex data assoicated with the shapes as a list
+        where each element of the list corresponds to one shape. Passing a
+        `shape_type` argument leads to only that particular `shape_type`
+        being returned.
+
+        Parameters
+        ----------
+        shape_type : {'line', 'rectangle', 'ellipse', 'path', 'polygon'} |
+                     None, optional
+            String of shape type to be included.
+
+        Returns
+        ----------
+        data : list
+            List of shape data where each element of the list is an
+            `np.ndarray` corresponding to one shape
+        """
+        if self.ndim == 2:
+            # For 2D shapes just convert current view to a list
+            data = self._data_view.to_list(shape_type=shape_type)
```

Comment:
I'm kinda surprised that 2D requires a special case here... Doesn't the nd code with the `()` key just work?

Reply:
I've reduced one less if statement here, but lumped the 2D case in with the broadcast code as it is simpler to understand I think and less complicated